cl/corpus_importer/utils.py

```diff
@@ -1,8 +1,11 @@
 from datetime import date
-from typing import Optional
+from typing import Any, Optional
 
 from django.utils.timezone import now
+from sklearn.feature_extraction.text import TfidfVectorizer
+from sklearn.metrics.pairwise import cosine_similarity
```

Comment:
The only imports of utils.py are in:

- cl/corpus_importer/tasks.py
- cl/corpus_importer/tests.py
- cl/corpus_importer/commands/harvard_merge.py
- cl/corpus_importer/commands/upload_to_ia.py
- cl/recap/mergers.py
- cl/recap/tasks.py

I used the call hierarchy tool in pycharm to track usage of the imported functions from utils.py but based on that, it looks good to me.

Reply:
Hm, but this means celery will be importing utils, which will import sklearn, which will up its RAM needs, right?

Reply:
I guess that would mean that your code would be importing it over and over, which isn't great either. Want to do a little research on this and see if there are any better solutions?

Reply:
According to documentation python caches modules, every module that is imported is stored in a dictionary called sys.modules, it maps the name of each module to the module object so when you import again the module python give us back the same module object instead of searching for the module and loading it again.

Also if we put the import inside the function, it will be loaded only when we execute the function, if we don't call to that function in celery service, the module won't be loaded.

https://docs.python.org/3.11/reference/import.html#the-import-system
https://docs.python.org/3.11/reference/import.html#the-module-cache

Reply:
Fantastic. Let's absolutely do this. I learned something new.